osu.Game/Screens/Play/HUD/ArgonHealthDisplay.cs

```diff
@@ -0,0 +1,284 @@
+            }, out resetMissBarDelegate);
+
+            missBar.FadeIn(120, Easing.OutQuint);
+            missBar.Delay(500).FadeOut(300, Easing.InQuint);
+
+            missBar.TransformTo(nameof(BarPath.BarColour), miss_bar_colour.Lighten(0.1f))
+                   .TransformTo(nameof(BarPath.BarColour), miss_bar_colour, 300, Easing.OutQuint);
+
+            missBar.TransformTo(nameof(BarPath.GlowColour), miss_bar_flash_colour)
+                   .TransformTo(nameof(BarPath.GlowColour), miss_bar_glow_colour, 300, Easing.OutQuint);
+        }
+
+        protected override void Flash(JudgementResult result)
+        {
+            base.Flash(result);
+
+            healthBar.TransformTo(nameof(BarPath.GlowColour), health_bar_flash_colour)
+                     .TransformTo(nameof(BarPath.GlowColour), health_bar_glow_colour, 300, Easing.OutQuint);
+        }
+
+        private double missBarValue = 1.0;
+        private readonly List<Vector2> missBarVertices = new List<Vector2>();
+
+        public double MissBarValue
+        {
+            get => missBarValue;
+            set
+            {
+                if (missBarValue == value)
+                    return;
+
+                missBarValue = value;
+                updatePathVertices();
+            }
+        }
+
+        private double healthBarValue = 1.0;
+        private readonly List<Vector2> healthBarVertices = new List<Vector2>();
+
+        public double HealthBarValue
+        {
+            get => healthBarValue;
+            set
+            {
+                if (healthBarValue == value)
+                    return;
+
+                healthBarValue = value;
+                updatePathVertices();
+            }
+        }
+
+        private void updatePathVertices()
+        {
+            barPath.GetPathToProgress(healthBarVertices, 0.0, healthBarValue);
+            barPath.GetPathToProgress(missBarVertices, healthBarValue, Math.Max(missBarValue, healthBarValue));
+
+            if (healthBarVertices.Count == 0)
+                healthBarVertices.Add(Vector2.Zero);
+
+            if (missBarVertices.Count == 0)
+                missBarVertices.Add(Vector2.Zero);
+
+            missBar.Vertices = missBarVertices.Select(v => v - missBarVertices[0]).ToList();
+            missBar.Position = missBarVertices[0];
+
+            healthBar.Vertices = healthBarVertices.Select(v => v - healthBarVertices[0]).ToList();
+            healthBar.Position = healthBarVertices[0];
+        }
+
+        private partial class BackgroundPath : SmoothPath
+        {
+            protected override Color4 ColourAt(float position)
+            {
+                if (position <= 0.128f)
+                    return Color4.White.Opacity(0.5f);
+
+                position -= 0.128f;
+                return Interpolation.ValueAt(Math.Clamp(position, 0f, 1f), Color4.White.Opacity(0.5f), Color4.Black.Opacity(0.5f), -0.75f, 1f, Easing.OutQuart);
+            }
```

Comment:
Removing position subtraction will probably change nothing in how jank this looks. The source design uses some sort of blurred inner shadow with a radius of 10px, and I'm not totally sure how this can be applied in here. The only way I can make this less jank would be by altering the interpolation start/end colours away from what they are in the source design, while achieving the same visual result.